src/detect-parse.c

```diff
@@ -708,6 +708,12 @@ static int SigParseOptions(DetectEngineCtx *de_ctx, Signature *s, char *optstr,
                     optname, optstr);
             goto error;
         }
+    } else if (st->flags & SIGMATCH_NOOPT) {
+        if (optvalue && strlen(optvalue)) {
+            SCLogError(SC_ERR_INVALID_SIGNATURE, "unexpected option to %s keyword: \'%s'", optname,
```

Comment:
I don't think `'` needs to be escaped here?

Reply:
Yeah, weird, I dunno where that came from, will fix it

Reply:
Yeah, it came from copy and pasting the above check. Fixed it in both places.